cloverage/test/cloverage/instrument_test.clj

```diff
@@ -37,17 +32,46 @@
   (method [_] foo))
 
 (t/deftest test-form-type
-  (t/is (= :atomic (form-type- 1)))
-  (t/is (= :atomic (form-type- "foo")))
-  (t/is (= :atomic (form-type- 'bar)))
-  (t/is (= :coll (form-type- [1 2 3 4])))
-  (t/is (= :coll (form-type- {1 2 3 4})))
-  (t/is (= :coll (form-type- #{1 2 3 4})))
-  (t/is (= :coll (form-type- (Record. 1))))
-  (t/is (= :list (form-type- '(+ 1 2))))
-  (t/is (= :do (form-type- '(do 1 2 3))))
-  (t/is (= :list (form-type- '(loop 1 2 3)
-                             {'loop 'hoop})))) ;fake a local binding
+  (doseq [[message form->expected]
+          {"Atomic forms"
+           {[1]     :atomic
+            ["foo"] :atomic
+            ['bar]  :atomic}
+
+           "Collections"
+           {[[1 2 3 4]]   :coll
+            [{1 2 3 4}]   :coll
+            [#{1 2 3 4}]  :coll
+            [(Record. 1)] :coll}
+
+           "do & loop"
+           {['(do 1 2 3)]               :do
+            ;; fake a local binding
+            '[(loop 1 2 3) {loop hoop}] :list}
+
+           "Inlined function calls"
+           {['(int 1)]            :inlined-fn-call
+            [`(int 1)]            :inlined-fn-call
+            [`(int 1) '{int int}] :inlined-fn-call}
+
+           "Local vars overshadow inlined fn calls; make sure this is detected correctly"
+           {'[(int) {int int}] :list}
+
+           "If some arities are inlined and others are not, only return `:inlined-fn-call` when inlined"
+           {['(+ 1)]   :list
+            ['(+ 1 2)] :inlined-fn-call}
+
+           "Lists starting with other lists"
+           {['((or + 1) 1 2)] :list}
+
+           "List-like classes e.g. clojure.lang.Cons for which `list?` is false"
+           {[(cons 1 '(2 3))] :list}}
+
+          [[form env] expected] form->expected]
+    (t/testing message
+      (t/testing (str "\n" (pr-str (list 'form-type form env)))
+        (t/is (= expected
+                 (inst/form-type form env)))))))
```

Comment:
Reworked this test so it uses `doseq` and adds `testing` context so it's easier to understand which form failed. Added new tests for `:inlined-fn-call` and some other stuff that was missing like lists starting with other lists